### ISSUE TITLE
Emojis break boxes [bug]

### ISSUE THREAD

**Issue by Kycermann** [chjj/blessed#422]
Emojis seem to have double (?) the width of normal characters. This isn't accounted for in a box when using `.pushLine`.

```┌──────────────────────────────┐
│Test 🐢                        │
│                              │ 
│                              │                                                
│                              │                                                
```

**Comment by asnewman** [comment]
@Kycermann some emojis have two character lengths. You can see this by doing `"🐢".length` (output would be 2)

**Comment by Kycermann** [comment]
That's true but they shouldn't distort the Blessed boxes they're in and at the moment they do

**Comment by p0ut** [comment]
@Kycermann This bug has been finally patched in a fork by kenan. 
[here]([https://github.com/kenan238/reblessed)

**Comment by Kycermann** [comment]
@p0ut awesome, seems like `here` doesn't have a link though

**Comment by p0ut** [comment]
@Kycermann Apologies, [https://github.com/kenan238/reblessed](https://github.com/kenan238/reblessed)

**Comment by p0ut** [comment]
@Kycermann So just a quick update so I don't mislead you, both myself and kenan are looking into the emoji issue. It is caused by the library not being able to detect the width of emoji's. Initially, we managed to 'semi-patch' such characters, however it still occurs on certain emoji's.

It seems that most character ranges are hard coded. We are doing our best to try to find a sufficient fix for this and I will keep you updated here.

**Comment by Kycermann** [comment]
@p0ut cheers, I appreciate it!

**Comment by nopublicsounds** [comment]
https://github.com/nopublicsounds/discord-tui/blob/main/tui/utils/unicodePatch.ts

wrote a monkey-patch function that replaces blessed's internal text-width calculation with the string-width library.
i struggled with this problem, didn't know that there was re-belssed..
